main.go

```diff
@@ -209,39 +492,104 @@ func runClient(host string) {
 		Transport: tr,
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	r := &netperfReader{doneCh: ctx.Done()}
-	r.buf = make([]byte, 1*humanize.MiByte)
-	rand.Read(r.buf)
+	var index string
+	for {
+		time.Sleep(1 * time.Second)
+		index = getWriterIndex(r, clnt)
+		if index == "" {
```

Comment:
```suggestion
		if index != "" {
			break
		}
```
... remove else section.